include/sparrow/typed_array.hpp

```diff
@@ -467,7 +467,7 @@ namespace sparrow
     bool operator==(const typed_array<T, Layout>& ta1, const typed_array<T, Layout>& ta2)
     {   
         // see https://github.com/man-group/sparrow/issues/108
-#if defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 180000)
+#if EMSCRIPTEN ||( defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 180000))
```

Comment:
It's possible that you won't need that if `<version>` is included.

Reply:
actualy its not needed at all.
I confused an error from the tests.

In the tests we need
```
  #ifndef EMSCRIPTEN
  constexpr bool valid = std::contiguous_iterator<test_iterator>;
  CHECK(valid);
  #endif
```
otherwise we see a similar error to the one we saw

Reply:
Yes I think it's better to not have this check (also it should be `defined(EMSCRIPTEN)`?)

Reply:
It seems this was removed and re-added. Can we confirm that it works without this check if `<version>` is included?

Reply:
I tried with including version https://github.com/man-group/sparrow/pull/130/commits/19cb95bf2a0ac4f31fe513f363349ddf4ad718ec but this failed

Reply:
Ah it seems that _LIBCPP_VERSION is not defined with Emscripten.

Reply:
Hmm ok then `EMSCRIPTEN` is probably the right define do check I suppose.

Reply:
Do we know if a clang18-based emscripten would also require this?

Reply:
> Do we know if a clang18-based emscripten would also require this?

we don't know. Atm we use the emscripten version of emscripten-forge (3.1.45).

Reply:
It's only to know if there should be an upper bound in these checks.